src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
-        const messages = this.props.conversation.messages.map((message) => {
+        const processedMessages = [...this.props.conversation.messages];
+        let lastAuthor;
+        processedMessages.forEach((message, index) => {
+            const author = message.role === 'appUser' ? message.role : `${message.role}_${message.name}`;
+
+            if (!lastAuthor) {
+                lastAuthor = author;
+                message.firstInGroup = true;
```

Comment:
Does that kind of stuff really belong in the reducer? It's not really data, it's more like markup used for presentation, so belongs more in the component, no? In fact that markup doesn't even really need to be persisted anywhere... during the `.map` loop [here](https://github.com/smooch/smooch-js/pull/267/files#diff-44044b6ef5eb6631027b51708c51c608R90), you can just look at the previous and next message to know how you should render the current one, there's no need to store those boolean flags anywhere other than in the props of the `MessageComponent`